server/public/model/client4_test.go

```diff
@@ -149,3 +153,30 @@ func TestClient4RequestCancellation(t *testing.T) {
 		<-done
 	})
 }
+
+func ExampleClient4_GetUsers() {
+	client := model.NewAPIv4Client("http://localhost:8065")
+	client.SetToken(os.Getenv("MM_TOKEN"))
+
+	const perPage = 100
+	var page int
+	var allUsers []*model.User
+	for {
+		users, _, err := client.GetUsers(context.TODO(), page, perPage, "")
+		if err != nil {
+			log.Printf("error fetching users: %v", err)
+			return
+		}
+
+		if len(users) == 0 {
+			break
+		}
+
+		allUsers = append(allUsers, users...)
```

Comment:
I sort of understand why we are doing this in the context of the example but in most cases I'd consider this to be an anti-pattern, having to paginate only to then have all the data in a single object. I'd expect this loop to print users as it's getting them.

Reply:
Good point. I agree that it's an anti-pattern but a commonly used one.

Anyway, we don't need to promote it. Moved the print into the loop.